docs/how_to_publish_to_cmr_with_earthdata_varinfo.ipynb

```diff
@@ -0,0 +1,314 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Instantiate a VarInfoFromNetCDF4 object for a local NetCDF-4 file.\n",
+    "var_info = VarInfoFromNetCDF4('MERRA2_400.inst1_2d_asm_Nx.20220130.nc4',\n",
+    "                              short_name='M2I1NXASM')"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Retrieve a dictionary of UMM-Var JSON records"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Keys are the full variable paths, values are \n",
+    "# UMM-Var schema-compatible, JSON-serialisable dictionaries.\n",
+    "umm_var_dict = get_all_umm_var(var_info)\n",
+    "umm_var_dict"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Publish all the UMM-Var entries for M2I1NXASM"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "publish_all_umm_var('C1256535511-EEDTEST',\n",
+    "                    umm_var_dict,\n",
+    "                    auth_header=auth_header,\n",
+    "                    cmr_env=CMR_UAT)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Publish a single UMM-Var entry"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Keys are the full variable paths, values are \n",
+    "# UMM-Var schema-compatible, JSON-serialisable dictionaries.\n",
+    "umm_var_entry = get_umm_var(var_info, var_info.get_variable('/TROPPV'))\n",
+    "umm_var_entry"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Retrieve a dictionary of a single UMM-Var JSON records. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "publish_umm_var('C1256535511-EEDTEST',\n",
+    "                umm_var_entry,\n",
+    "                auth_header=auth_header,\n",
+    "                cmr_env=CMR_UAT)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Download a granule for the collection **GLDAS_NOAH10_3H**, generate it's UMM-Var entry and publish it to CMR, with our one function: ```generate_collection_umm_var```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add publish=True if you want publish directly to CMR UAT\n",
+    "generate_collection_umm_var(collection_concept_id='C1256543837-EEDTEST',\n",
+    "                            auth_header=auth_header)"
```

Comment:
It'd be good to include some truncated expected output in the markdown cell above this call - just so the user knows what to expect, and if the code ran correctly. (I think that's applicable to a lot of the code cells in the notebook)

Reply:
* [ ] Add truncated output